### PR TITLE
Refresh USS Folder Contents

### PR DESCRIPTION
**Pull request by rudyflores**
## Proposed changes

Prior to this PR the user was unable to refresh file names in a USS directory without having to collapse the tree, this could be unintuitive for the user since the tree would collapse and would have to expand their directories after every refresh is performed. With this new change the user will be able to refresh directory file names without the tree collapsing and provide a better experience for the user.

Screen Shot 2021-06-21 at 12 34 13 PM

## Release Notes

Milestone: 1.17
Changelog: Refresh for USS directory file names without the tree collapsing

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [x] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Review by JillieBeanSim — CHANGES_REQUESTED** [review]
@rudyflores functionality works great! The only thing I see that will need fixing is the broken unit tests which just need the command count updated like done in the DS Refresh PR
image

File: packages/zowe-explorer/package.json [hunk 1]
```diff
@@ -699,6 +708,16 @@
           "command": "zowe.uss.removeSavedSearch",
           "group": "inline"
         },
+        {
+          "when": "view == zowe.uss.explorer && viewItem =~ /^(?!(directory|ussSession|favorite|profile_fav))/",
+          "command": "zowe.uss.refreshUSS",
+          "group": "0_mainframeInteraction"
+        },
```
Comment (lauren-li) [inline packages/zowe-explorer/package.json:715]: @rudyflores Please remove this block (lines 711-715), as `zowe.uss.refreshUSS` is already declared with the same `when` clause further down (lines 736-740). It is causing the refresh ("Pull from mainframe") command to appear twice for files:

Screen Shot 2021-06-23 at 10 47 35 AM

**Review by lauren-li — CHANGES_REQUESTED** [review]
This PR is off to a good start! Once you address my comment below in addition to @JillieBeanSim's earlier request to update the unit test, I will review the PR again. Thanks @rudyflores!

**Review by lauren-li — DISMISSED** [review]
This looks good to me now. All unit and expected integration tests are passing. The new command is located in the correct context menu group, and functions as expected. Thanks @rudyflores for addressing my comments!

**Review by JillieBeanSim — COMMENTED** [review]
@rudyflores this LGTM, thanks for addressing the comments and for your work on this.  

I am wondering if we should also have this option at the session node to refresh the search if something was done at top level.  For example:
do a USS filter search for `/u/userid/directory` then something in that directory changes on host side, like a new file or a file was deleted/renamed, should the option to `Refresh USS Directory` appear in right-click menu here as well?  @lauren-li 
image

**Review by JillieBeanSim — APPROVED** [review]
LGTM! Thanks @rudyflores this is an awesome feature.

**Comment by phaumer** [comment]
Very nice. I cannot believe how easy that was :-) 

I would change the name though: as we have "Create Directory" directly below the naming should be consistent and just be  "Refresh Directory" without the USS; the entire view is USS.

**Review by lauren-li — APPROVED** [review]
This PR looks good to me. All unit and expected integration tests are still passing, and the added functionality to refresh the USS directory from the profile level is working well.

I am approving this PR, but I also think @phaumer made a good point that the command titles will be more consistent if we use "Refresh Directory" instead of "Refresh USS Directory". I would be happy to re-approve the PR with this change being made.

Thanks @rudyflores for your hard work on this PR!

**Review by lauren-li — APPROVED** [review]
This PR looks good to me. Thanks @rudyflores for the command label update!

**Review by JillieBeanSim — APPROVED** [review]
LGTM, thanks @rudyflores

**Pull request merged**